Add iconDir to SearchResultCardData

labkey-alan opened the pull request:
#### Rationale
We are adding notebooks to full text search in Biologics, and we need to render a notebook icon, but it does not live in the default directory used by SearchResultCard. This PR adds a new optional attribute to SearchResultCardData that lets apps override this directory.

#### Related Pull Requests
* https://github.com/LabKey/labbook/pull/148
* https://github.com/LabKey/biologics/pull/1193

#### Changes
* Add iconDir to SearchResultCardData
  * This allows apps to render icons in search results that are in non-default directories
* Make URLResolver.resolveSearchUsingIndex synchronous
  * It was a promise, but it did nothing async

labkey-alan committed:
Make URLResolver.resolveSearchUsingIndex synchronous

It was unnecessarily a promise
Improved typing

labkey-alan committed:
Add optional iconDir to SearchResultCard model

labkey-alan committed:
Update version and release notes

cnathe commented on packages/components/src/internal/components/search/SearchResultCard.tsx:
```diff
@@ -57,8 +57,9 @@ export class SearchResultCard extends React.Component<SearchResultProps, any> {
+        const iconDir = cardData.iconDir ?? '_images';
 
-        return <SVGIcon iconDir="_images" iconSrc={iconSrc} alt={altText} className="search-result__card-icon" />;
+        return <SVGIcon iconDir={iconDir} iconSrc={iconSrc} alt={altText} className="search-result__card-icon" />;
```
is there a jest test that could be added for this new prop?

labkey-alan replied:
I refactored the existing tests (no more snapshots!) and added more test cases in my most recent push.

labkey-nicka commented on packages/components/src/internal/components/search/SearchResultCard.tsx:
```diff
@@ -57,8 +57,9 @@ export class SearchResultCard extends React.Component<SearchResultProps, any> {
         }
 
         const { iconSrc, altText } = cardData;
+        const iconDir = cardData.iconDir ?? '_images';
```
This actually isn't necessary to set as the default is already "_images". I would just pass along `iconDir`.

labkey-alan replied:
I fixed this and a few other usages as well.

labkey-alan committed:
SVGIcon: Update usages to not pass default value

labkey-alan committed:
Convert SearchResultCard to FC

labkey-alan committed:
Refactor SearchResultCard tests

No longer use snapshot tests
Add cases for iconDir
Add cases for summary
Add cases for typeName

labkey-alan committed:
Prep for release
